hazelcast/src/main/java/com/hazelcast/map/impl/operation/ReplaceAllOperation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.spi.impl.operationservice.MutatingOperation;
+
+import java.util.function.BiFunction;
+
+public class ReplaceAllOperation<K, V, R> extends MapOperation implements MutatingOperation {
```

Comment:
I think we don't need to introduce a new operation. It is easy and less error prone to implement this functionality by using entry processor mechanism. To see an example implementation please follow the client side implementation of [removeAll](https://github.com/hazelcast/hazelcast/blob/fb054c67f1cc8240f0bfedb941761d12b4503e9f/hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java#L313). It makes use of entry processor on server side, in your case which is already available as `MapEntryReplacingEntryProcessor`. You just need to send your function to server and server will handle the rest.

Reply:
Done